### PR TITLE
build system: remove the config/ directory

### PR DESCRIPTION
**Pull request by shindere**
None of the files in this directory is used any more.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -180,11 +166,7 @@ tools/markdown-add-pr-links.sh text eol=lf
 runtime/caml/m.h.in text eol=lf
 runtime/caml/s.h.in text eol=lf
 
-# These two are cat scripts, so may not actually require this
-config/auto-aux/sharpbang text eol=lf
-config/auto-aux/sharpbang2 text eol=lf
-
-# Similarly, these are all Perl scripts, so may not actually require this
+# Tthese are all Perl scripts, so may not actually require this
```
Comment (dra27) [inline .gitattributes]:
```suggestion
# These are all Perl scripts, so may not actually require this
```
(typo on `These`)

File: HACKING.adoc [hunk 2]
```diff
@@ -179,13 +179,13 @@ has excellent documentation.
   VERSION::               version string
   asmcomp/::              native-code compiler and linker
   boot/::                 bootstrap compiler
+  build-aux/:             autotools suport scripts
```
Comment (dra27) [inline HACKING.adoc]:
```suggestion
  build-aux/:             autotools support scripts
```
(another typo)

**Review by dra27 — COMMENTED** [review]
Some typos. I'd suggest completely reverting the changes to `configure.ac` (and so `configure`). It's better to keep the full path as `git log -- config/m-nt.h` will still work after this PR is merged. At some point in the future, the entire comment will be superfluous and can be deleted.

**Comment by shindere** [comment]
Thanks a lot, @dra27.

The branch has been rebased and I believe that I have taken into account
all your comments.

**Comment by dra27** [comment]
Indeed, thanks @shindere!

**Comment by dra27** [comment]
Cherry-picked to 4.09: 405cff3729c4f1f88785e92f663a6d9e9808988c

**Comment by shindere** [comment]
David Allsopp (2019/06/10 05:57 -0700):
> Merged #8720 into trunk.

Thanks a lot, @dra27!

**Comment by shindere** [comment]
David Allsopp (2019/06/10 06:00 -0700):
> Cherry-picked to 4.09: 405cff3729c4f1f88785e92f663a6d9e9808988c

Thanks for that, too. I wasn't sure but happy you made the decision. :)

**Comment by dra27** [comment]
@shindere:
> David Allsopp (2019/06/10 06:00 -0700):
>  Cherry-picked to 4.09: 405cff3
> 
> Thanks for that, too. I wasn't sure but happy you made the decision. :)

It made sense that they were left around for 4.08, but there's certainly already been one instance of someone not realising that they're not supposed to be being used anymore, so better to eliminate the confusion!

**Comment by shindere** [comment]
David Allsopp (2019/06/10 06:10 -0700):
> @shindere:
> > David Allsopp (2019/06/10 06:00 -0700):
> >  Cherry-picked to 4.09: 405cff3
> > 
> > Thanks for that, too. I wasn't sure but happy you made the decision. :)
> 
> It made sense that they were left around for 4.08, but there's
> certainly already been one instance of someone not realising that
> they're not supposed to be being used anymore, so better to eliminate
> the confusion!

+1!

**Pull request merged**